modules/javafx.graphics/src/main/native-prism-sw/JAbstractSurface.c

```diff
@@ -85,7 +85,7 @@ Java_com_sun_pisces_AbstractSurface_getRGBImpl(JNIEnv* env, jobject objectHandle
         y < 0 || y >= surfaceHeight ||
         width  < 0 || width  > (surfaceWidth  - x) ||
         height < 0 || height > (surfaceHeight - y) ||
-        scanLength < width) {
+        scanLength < width || offset < 0) {
```

Comment:
this might be a naive question: would it make more sense to fail earlier, i.e. to test the offset before getting the surface pointer?

Reply:
No, that would be a larger change and out of scope. More importantly, the order of the checks doesn't matter.